test/h-group/level-11.js

```diff
@@ -0,0 +1,299 @@
+import { strict as assert } from 'node:assert';
+import { describe, it } from 'node:test';
+
+import { PLAYER, VARIANTS, setup, takeTurn } from '../test-utils.js';
+import * as ExAsserts from '../extra-asserts.js';
+import HGroup from '../../src/conventions/h-group.js';
+import { ACTION, CLUE } from '../../src/constants.js';
+import logger from '../../src/tools/logger.js';
+import { take_action } from '../../src/conventions/h-group/take-action.js';
+import { find_clues } from '../../src/conventions/h-group/clue-finder/clue-finder.js';
+
+logger.setLevel(logger.LEVELS.ERROR);
+
+describe('bluff clues', () => {
+	it('understands a direct play if the bluff isn\'t played into', () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['p1', 'y5', 'b1', 'g5', 'p2'],
+			['b3', 'r1', 'b5', 'b2', 'y4']
+		], {
+			level: 11,
+			play_stacks: [2, 2, 2, 2, 2],
+			starting: PLAYER.BOB
+		});
+		takeTurn(game, 'Bob clues red to Alice (slot 2)');
+
+		// Cathy's slot 1 could be any of the playable 3's.
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.CATHY][0].order].finessed, true);
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.CATHY][0].order], ['r3', 'y3', 'g3', 'b3', 'p3']);
+		// Alice's slot 2 could be r3 or r4.
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.ALICE][1].order], ['r3', 'r4']);
+
+		takeTurn(game, 'Cathy discards y4 (slot 5)', 'y1');
+
+		// After Cathy doesn't play into it, assume we have a play. 
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.ALICE][1].order], ['r3']);
+	});
+
+	it('understands giving a direct play through a bluff opportunity', () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['b3', 'r1', 'b1', 'g5', 'p2'],
+			['p1', 'r3', 'b5', 'b2', 'y4']
+		], {
+			level: 11,
+			play_stacks: [2, 2, 2, 2, 2],
+			starting: PLAYER.ALICE
+		});
+		takeTurn(game, 'Alice clues red to Cathy (slot 2)');
+
+		// Bob's slot 1 could be any of the playable 3's.
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.BOB][0].order].finessed, false);
+		// Cathy's slot 2 will be known to be r3 or r4 until Bob reacts.
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.CATHY][1].order], ['r3']);
```

Comment:
The comment here says that Cathy's slot 2 is [r3,r4], but the code only checks for r3. I'm not sure which was the intention, but it seems like it should be [r3,r4].

Reply:
This seems to be consistent with how clues that look like finesses are tracked, every player who knows notes the eventual result. I agree it feels like it should be [r3, r4], but it should be the same as regular finesses, so if we want to change this I'd expect to do it for both. I suspect it's better for everyone to track what will be known so that they don't worry about the other possibility for future clues.